provisioning_templates/snippet/preseed_networking_setup.erb

```diff
@@ -7,12 +7,41 @@ description: this will configure your host networking, it configures your primar
     as other configures NICs. It supports physical, VLAN and Alias interfaces. It's intended to be
     called in your preseed finish template.
 %>
+<% is_ubuntu = @host.operatingsystem.name == 'Ubuntu' -%>
+<% os_major = @host.operatingsystem.major.to_i -%>
 <% host_subnet = @host.subnet -%>
 <% host_dhcp = host_subnet.nil? ? false : host_subnet.dhcp_boot_mode? -%>
 <% host_subnet6 = @host.subnet6 -%>
 <% host_dhcp6 = host_subnet6.nil? ? false : host_subnet6.dhcp_boot_mode? -%>
 
 real=`ip -o link | awk '/<%= @host.mac -%>/ {print $2;}' | sed s/://`
+<%# Ubuntu 18.04 and newer uses netplan instead of /etc/network/interfaces -%>
+<% if is_ubuntu && os_major >= 18 -%>
+cat << EOF > /etc/netplan/01-netcfg.yaml
+# This file describes the network interfaces available on your system
+# For more information, see netplan(5).
+network:
+  version: 2
+  renderer: networkd
```

Comment:
Is this sane default for Ubuntu? I mean, in Fedora world the network daemon of choice is NetworkManager.

Reply:
https://blog.ubuntu.com/2017/07/05/quick-and-easy-network-configuration-with-netplan suggests networkd is the default but I don't know if that's the final version.

Reply:
They elaborate on some wifi limitation tho, can you please make this a host parameter? Use any default you want, but user should be able to redefine this.

Reply:
I believe that `networkd` is used as the default renderer, and specifying it is not necessary. 

The 18.04 manpage at https://manpages.ubuntu.com/manpages/bionic/man5/netplan.5.html says the following, and this should be the official word:

>        · Configured  devices get handled by systemd-networkd by default, unless explicitly marked
>         as managed by a specific renderer (NetworkManager)